server/src/compression.rs

```diff
@@ -566,6 +637,110 @@ mod tests {
+    ///
+    /// # Arguments
+    ///
+    /// * `length` - An unsigned integer that denotes how many data-points should be generated.
+    /// * `linear` - A boolean that denotes if the data should be linear.
+    /// * `random` - A boolean that denotes if the data should be randomized.
+    /// * `min_step` - Denotes the minimum change that should be applied from one data point to the next when randomizing data. Only used if `random` is set to `true`.
+    /// * `max_step` - Denotes the maximum change that should be applied from one data point to the next when randomizing data. Only used if `random` is set to `true`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let constant_data = generate_data(100, false, false, None, None);
+    /// let random_data = generate_data(100, false, true, Some(10.0), Some(20.0));
+    /// let linear_data = generate_data(100, true, false, None, None);
+    /// let random_linear = generate_data(100, true, true, Some(9.0), Some(11.0));
+    /// ```
+    pub fn generate_data(
+        length: u16,
+        linear: bool,
+        random: bool,
+        min_step: Option<f32>,
+        max_step: Option<f32>,
+    ) -> Vec<f32> {
+        let mut randomizer = thread_rng();
+        let mut values: Vec<f32> = vec![];
+
+        if random {
+            if min_step == None || max_step == None {
+                panic!("min_step and max_step need to be entered if random_values is true.")
+            } else if min_step > max_step {
+                panic!("min_step cannot be lower than the max_step.")
+            }
+        }
+
+        match (linear, random) {
+            (true, true) => {
+                let mut random_linear = vec![];
+                let mut previous_value: f32 = 0.0;
+                for _ in 0..length {
+                    let next_value = (randomizer
+                        .sample(Uniform::from(min_step.unwrap()..max_step.unwrap())))
+                        + previous_value;
+                    random_linear.push(next_value);
+                    previous_value = next_value;
+                }
+                values.append(&mut random_linear);
+            }
+            (true, false) => {
+                let mut linear =
+                    Vec::from_iter((10..(length + 1) * 10).step_by(10).map(|v| v as f32));
+                values.append(&mut linear);
+            }
+            (false, true) => {
+                let mut random = vec![];
+                for _ in 0..length {
+                    random.push(
+                        randomizer.sample(Uniform::from(min_step.unwrap()..max_step.unwrap())),
+                    );
+                }
+                values.append(&mut random);
+            }
+            (false, false) => {
+                let mut constant =
+                    vec![randomizer.sample(Uniform::from(0.0..f32::MAX)); length as usize];
```

Comment:
True, but I think that it was also confusing that the `generate_data` function was used in some cases, and in other cases, the data was hard coded. Don't you think? 

It is also useful for generating a vector of values with a certain length. The other solution would be to separate it into two functions, `generate_random_data` and `generate_data`, but that seems more convoluted than beneficial, right?

Reply:
Can you not just change it to return the same value (for example 10.0) `length` times when `false, false` is chosen. Quickly looking at the functions that use `generate_data` it seems that this would not break anything. Then it would be constant and not random.

Reply:
Sure, seems fine to me. Fixed.